### PR TITLE
feat(runtime, ci): finish migration to bare-metal benchmark runner and update weights

### PR DESCRIPTION
**Pull request by ByteNacked**
### What was done:

A migration of the benchmark runner was performed, moving from a virtual host to a bare-metal host to increase the reproducibility of benchmark results.

Since the bare-metal host differs somewhat in performance from the equivalent virtual host, we investigated these differences through several benchmark runs and calculated the average discrepancy between them. As a result, a 10% adjustment was introduced to the instruction weights, aligning the instruction weight results with those on the virtual host.

Additionally, for the multi-threaded benchmarks of the `gear_builtin` pallet, it was decided to fix the execution of these benchmarks to 4 CPU cores, again to improve the reproducibility of the results.

* [x] Added script which adjust Instruction weights by 10 % when running on bare metal runner
* [x] Run multi threaded benches (`pallet_gear_builtin`) on fixed number of cores (4 cores)
* [x] Update weights from bare-metal run

@reviewer-or-team

### CODE REVIEW

File: pallets/gear-builtin/src/weights.rs [hunk 1]
```diff
@@ -19,7 +19,7 @@
 //! Autogenerated weights for pallet_gear_builtin
```
Comment (ByteNacked) [inline pallets/gear-builtin/src/weights.rs:19]: Please ignore `pallets/gear-builtin/src/weights.rs` results, it was run on the wrong number of cores.

Comment (ByteNacked) [reply pallets/gear-builtin/src/weights.rs:19]: Will be fixed soon

Comment (ByteNacked) [reply pallets/gear-builtin/src/weights.rs:19]: Fixed

**Comment by ByteNacked** [comment]
Should [`InstructionWeight` version](https://github.com/gear-tech/gear/blob/0c1fc4236e81f1d6a9bbd74c22dd0f38796afdee/pallets/gear/src/schedule.rs#L758)  be incremented in this PR due to some instruction weight change?

**Review by breathx — APPROVED** [review]

**Comment by breathx** [comment]
Please resolve master brach conflicts

**Pull request merged**